src/App.jsx

```diff
@@ -14,45 +15,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const sortTypes = {
+  ALPHABET: 'ALPHABET',
+  LENGTH: 'LENGTH',
+};
+
+const sortHandlers = {
+  ALPHABET: (a, b) => a.localeCompare(b),
+  LENGTH: (a, b) => a.length - b.length,
+};
+
+const getPreparedGoods = (listOfGoods, sortType, isReversed) => {
+  const nextListOfGoods = [...listOfGoods];
+  const sortHandler = sortHandlers[sortType];
+
+  if (sortHandler) {
+    nextListOfGoods.sort(sortHandler);
+  }
+
+  if (isReversed) {
+    nextListOfGoods.reverse();
+  }
+
+  return nextListOfGoods;
+};
+
+export const App = () => {
+  const [sortType, setSortType] = useState(null);
+  const [isReversed, setIsReversed] = useState(false);
+  const listOfGoods = getPreparedGoods(goodsFromServer, sortType, isReversed);
+
+  const isResetDisplayed = listOfGoods.toString() !== goodsFromServer.toString()
+    || sortType !== null
+    || isReversed;
```

Comment:
it seems that you twice check the same condition, `listOfGoods !== goodsFromServer` only when `sortType` or `isReversed` return true